pkg/iac-providers/kustomize/v3/load-dir.go

```diff
@@ -0,0 +1,122 @@
+package kustomizev3
+
+import (
+	"errors"
+	"path/filepath"
+
+	k8sv1 "github.com/accurics/terrascan/pkg/iac-providers/kubernetes/v1"
+	"github.com/accurics/terrascan/pkg/iac-providers/output"
+	"github.com/accurics/terrascan/pkg/utils"
+	"go.uber.org/zap"
+	"sigs.k8s.io/kustomize/api/filesys"
+	"sigs.k8s.io/kustomize/api/krusty"
+)
+
+const (
+	kustomizedirectory string = "kustomize_directory"
+)
+
+// LoadIacDir loads the kustomize directory
+func (k *KustomizeV3) LoadIacDir(absRootDir string) (output.AllResourceConfigs, error) {
+
+	allResourcesConfig := make(map[string][]output.ResourceConfig)
+
+	files, err := utils.FindFilesBySuffixInCurrentDir(absRootDir, KustomizeFileNames())
+	if err != nil {
+		zap.S().Warn("error while searching for iac files", zap.String("root dir", absRootDir), zap.Error(err))
+		return allResourcesConfig, err
+	}
+
+	if len(files) == 0 {
+		err := errors.New("could not find a kustomization.yaml/yml file in the directory")
+		zap.S().Warn("error while searching for iac files", zap.String("root dir", absRootDir), zap.Error(err))
+		return allResourcesConfig, err
+	}
+
+	if len(files) > 1 {
+		err := errors.New("a directory cannot have more than 1 kustomization.yaml/yml file")
+		zap.S().Warn("error while searching for iac files", zap.String("root dir", absRootDir), zap.Error(err))
+		return allResourcesConfig, err
+	}
+
+	var config output.ResourceConfig
+	config.Type = kustomizedirectory
+	config.Name = filepath.Dir(absRootDir)
+	config.Line = 0
+	config.ID = config.Type + "." + config.Name
+
+	var yamlkustomizeobj map[string]interface{}
+	var kustomizeFileName string
+	for _, filename := range KustomizeFileNames() {
```

Comment:
Can we use `files` variable rather than calling `KustomizeFileNames`?

Reply:
Nice catch